Domain/Business/QueryServices/PointQueryServices/FetchFullPointsByUserIdQueryService.cs

```diff
@@ -0,0 +1,53 @@
+﻿using Domain.Data.Queries;
+using Domain.Data.Queries.PointReasonQueries;
+using Domain.Data.Queries.PointQueries;
+using Domain.Entities.Blog;
+using Domain.Entities.View;
+using EnsureThat;
+using System;
+using System.Collections.Generic;
+
+namespace Domain.Business.QueryServices.PointQueryServices
+{
+    public class FetchFullPointsByUserId : IQueryService<FetchFullPointsByUserIdQuery, IEnumerable<FullPoint>>
+    {
+        private readonly IQueryProcessor QueryProcessor;
+
+        public FetchFullPointsByUserId(IQueryProcessor queryProcessor)
+        {
+            EnsureArg.IsNotNull(queryProcessor, nameof(queryProcessor));
+            QueryProcessor = queryProcessor;
+        }
+
+        public IEnumerable<FullPoint> Execute(FetchFullPointsByUserIdQuery query)
+        {
+            var points = QueryProcessor.Execute(new FetchPointsByUserIdQuery
+            {
+                UserId = query.UserId,
+            });
+
+            return GetFullPoints(points);
+        }
+
+        private IEnumerable<FullPoint> GetFullPoints(IEnumerable<Point> points)
```

Comment:
Although this is a good use for a query service in the domain, but since this is a database class we should probably simplify this to just execute a stored procedure on the database. Also once that procedure is created in the database, lets say it's called something like:

```sql
CREATE OR ALTER PROCEDURE Blog.FetchFullPointsByUserId(
    @UserId
)
...
``` 

Then to use the automatic query generator, we would just have to make sure our Query POCO matches the name of the SQL procedure - the "Query" part. So the POCO object, to call the above example procedure name, would have to be called `FetchFullPointsByUserIdQuery` which it already it. Also the property names on the `FetchFullPointsByUserIdQuery` object would have to match the names of the input parameters to the procedure. So, the `FetchFullPointsByUserIdQuery` would have to contain a property named `UserId`. The automatic query generator will then generate a QueryService that will then effectively execute the following sql command: `EXEC Blog.FetchFullPointsByUserIdQuery @UserId {FetchFullPointsByUserIdQuery.UserId}`. Using this way you would also have to create a custom reader for `FullPoint`. I can walk you through this process if you would like.

Also, I would prefer writing a custom db procedure because we could use another interesting report query for our Project Report.

Reply:
I actually ended up using `GetPointBreakdownByUserIdQuery` instead of  `FetchFullPointsByUserIdQuery`. I will go ahead and create a neat custom procedure for `GetPointBreakdownByUserId` though.

Reply:
Yeah that will work, you can look at PR's #26 and #27 to see more of how the automatic db service generation works.